Tests/NIOPosixTests/VsockAddressTest.swift

```diff
@@ -0,0 +1,80 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2021 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+import XCTest
+@testable import NIOCore
+@testable import NIOPosix
+
+class VsockAddressTest: XCTestCase {
+
+    func testDescriptionWorks() throws {
+        XCTAssertEqual(VsockAddress(cid: .host, port: 12345).description, "[VSOCK]2:12345")
+        XCTAssertEqual(VsockAddress(cid: .any, port: 12345).description, "[VSOCK]-1:12345")
+        XCTAssertEqual(VsockAddress(cid: .host, port: .any).description, "[VSOCK]2:-1")
+        XCTAssertEqual(VsockAddress(cid: .any, port: .any).description, "[VSOCK]-1:-1")
+    }
+
+    func testInitializeFromIntegerLiteral() throws {
+        XCTAssertEqual(VsockAddress.ContextID(integerLiteral: 0), 0)
+        XCTAssertEqual(VsockAddress.Port(integerLiteral: 0), 0)
+        XCTAssertEqual(VsockAddress.ContextID(integerLiteral: 4294967295), 4294967295)
+        XCTAssertEqual(VsockAddress.Port(integerLiteral: 4294967295), 4294967295)
+    }
+
+    func testInitializeFromInt() throws {
+        XCTAssertEqual(VsockAddress.ContextID(0), 0)
+        XCTAssertEqual(VsockAddress.ContextID(4294967295), 4294967295)
+        XCTAssertEqual(VsockAddress.Port(0), 0)
+        XCTAssertEqual(VsockAddress.Port(4294967295), 4294967295)
+//        XCTAssertEqual(VsockAddress.ContextID(-1), 4294967295)  // compile-time error
```

Comment:
some comments left

Reply:
Left them there for posterity, but I removed now.